### ISSUE TITLE
Non-SHA commits under the cursor ignored by `fugitive_c`

### ISSUE THREAD

**Issue by PiCT0** [tpope/vim-fugitive#1485]
At 3c3e6ea67b0250db077e4a7c1b033f4537893e26, given the following example `:Gstatus`:
```
Head: master
Merge: origin/master

Staged (1)
M README.md

Unpushed to origin/master (1)
07acba2 <commit message>
```

I would like to create a `fixup!` commit for `origin/master`. From `:help fugitive_c`:
```
Commit maps

cc                      Create a commit.
[...]
cf                      Create a fixup! commit for the commit under the
                        cursor.
```

But `cf` with the cursor on `origin/master` (second line of `:Gstatus`) only gives the pre-populated prompt
```
:Git commit --fixup=
```
instead of the desired
```
:Git commit --fixup=origin/master
```
As the above command runs, isn't it sensible to want  `cf` to fully build it for the user?

Obviously, `cf` on the SHA (`07acba2`) gives the expected prompt:
```
:Git commit --fixup=07acba2
```

Now, I understand that `origin/master` is not a SHA but it still names a _commit_ (technically, a Git revision) as interpreted by Git. Therefore, shouldn't it also be accepted as a _commit_ by `fugitive`? Are there any limitations preventing it from identifying `origin/master` as a valid revision for `--fixup`?

If the desired behaviour can't be obtained, it should be made clear throughout the documentation by replacing occurrences of `commit` with `SHA` (or equivalent) in (at least) `:help fugitive_c`.

---

**NB**: I believe this issue might also concern `cF`, `cs`, `cS`, `cA`, `crc`, ...

**Comment by tpope** [comment]
Rewriting a commit that's already pushed to master is a really strange pitch, as it's a textbook example of when not to rebase. The big picture does include some less weird use cases such as revert on `crc`. It's still a bit awkward because there's no visible commit message, which means you're relying on your own memory of what that ref points to, and as such I never bothered to hook it up. But sure, if the user invokes the map on such a header line, may as well DWIM.

> If the desired behaviour can't be obtained, it should be made clear throughout the documentation by replacing occurrences of `commit` with `SHA` (or equivalent) in (at least) `:help fugitive_c`.

This is moot, but for the record, absolutely not. The old behavior wasn't "SHAs only, no symbolic names", it was "commits only, no headers, regardless of if those headers happen to contain a symbolic name".

**Comment by PiCT0** [comment]
> Rewriting a commit that's already pushed to master is a really strange pitch

True but it's basically like walking around in your underwear: It's weird unless you're at home, alone. This was being done in the very early/hacky stages of a personal (private) repository, in order to keep track of the prototyping/changes across multiple machines. Definitely dirty but nobody was supposed to know!

Ignoring that very specific case, it definitely is

> a textbook example of when not to rebase.

... and this illustrates how a badly chosen example has the potential to take the attention away from the big picture!

> The old behavior wasn't "SHAs only, no symbolic names", it was "commits only, no headers"

As an oblivious user, I hadn't even considered the fact that headers might have been a special case. This makes perfect sense.

>  @tpope closed this in 8388b6d0fb5d3031369fd77751d6087f5330b340

Thanks (and props) for addressing this so quickly.
